flink-runtime/src/main/java/org/apache/flink/runtime/jobmanager/scheduler/NoResourceAvailableException.java

```diff
@@ -19,40 +19,20 @@
 package org.apache.flink.runtime.jobmanager.scheduler;
 
 import org.apache.flink.runtime.JobException;
-import org.apache.flink.runtime.throwable.ThrowableAnnotation;
-import org.apache.flink.runtime.throwable.ThrowableType;
 
-@ThrowableAnnotation(ThrowableType.NonRecoverableError)
+/**
+ * Indicates resource allocation failures.
+ */
 public class NoResourceAvailableException extends JobException {
```

Comment:
I assume there is no semantical change if we remove `@ThrowableAnnotation(ThrowableType.NonRecoverableError)` because `ThrowableClassifier` was never really used, right?

Reply:
Yes. It's only used by the `ExecutionFailureHandler` in the new scheduler.